### ISSUE TITLE
custom_sell() does not sell intra-candle but sells only on open

### ISSUE THREAD

**Issue by ryhnm** [freqtrade/freqtrade#6509]
## Describe your environment

Operating system: macOS Monterey 12.2
Python Version: 3.9.7
CCXT version: 1.72.98
Freqtrade Version: 2022.1

## Describe the problem:

My strategy uses the `custom_sell()` function to have a fixed risk to reward ratio for exits. However during backtesting, I noticed that the exits are always on the open of a candle instead of intra-candle at the true value where it is supposed to sell the trade. The documentation says the `custom_sell()` function is checked roughly every 5 seconds so this should not be the case. 

### Steps to reproduce:

These are my configurations:

  `order_types = {
          'buy': 'market',
          'sell': 'market',
          'emergencysell': 'market',
          'stoploss': 'market',
          'stoploss_on_exchange': True
      }`

and this is my `custom_sell()` function:

  ```
def custom_sell(self, pair: str, trade: 'Trade', current_time: 'datetime', current_rate: float, current_profit: float, **kwargs):

        #Start at next second candle to let prices stabilise
        if current_time - datetime.timedelta(minutes=60) > trade.open_date_utc:
        
            dataframe, _ = self.dp.get_analyzed_dataframe(pair, self.timeframe)
            
            trade_date = timeframe_to_prev_date(self.timeframe, trade.open_date_utc)
            trade_candle = dataframe.loc[dataframe['date'] == trade_date]

            if not trade_candle.empty:

                stoploss_price = trade_candle['stoploss'].iloc[0]

                risk_distance = trade.open_rate - stoploss_price
                reward_distance = risk_distance*2

                take_profit_price_abs = trade.open_rate + reward_distance

                if (current_rate >= take_profit_price_abs):
                    return 'sell_profit'

                if (current_rate < stoploss_price):
                    return 'sell_loss'
```
  
### Observed Results:

Exits are always too late because the trade is never sold at the stoploss where it is supposed to sell because it only gets sold on an open of a candle instead of immediately against market.

![Screen Shot 2022-03-08 at 3 23 06 PM](https://user-images.githubusercontent.com/99715402/157229004-884b6ee0-706d-4660-aa5c-9b2a2fbcce9d.png)

**Comment by xmatthias** [comment]
During backtesting, freqtrade has 4 values per candle - open, high low and close.
all callbacks are therefore executed exactly once per candle during backtesting. 
The Statement about "roughly every 5 seconds" applies only to live - in backtesting, we would have to call it with the same exact value - which would provide no benefit.
the [freqtrade basics](https://www.freqtrade.io/en/stable/bot-basics/#backtesting-hyperopt-execution-logic) page also details this:

> Loops per candle simulating entry and exit points.


You can partially work around this by using `--timeframe-detail 1m` (assuming you have 1m data) - but the lowest you can get with this is 1m - so custom_sell is called once every minute - which is still once per  (detail) candle.

Using a "proper" stoploss (either via static stoploss, or "custom_stoploss") will set the value to the desired stop - and execute the sell at a different value than open should it be hit if that's what you're after. But especially with custom-stoploss, you'll need to be aware how it's working (Read the documentation) to provide the correct value.

**Comment by ryhnm** [comment]
I see, thanks for the response. So then I have 2 questions. First, since in live modus the function will get checked every 5 seconds it should behave as expected with these current settings right?

Second, when backtesting I noticed some of the exits have a sell reason called 'trailing_stop_loss'. How can this be when I only have the `custom_sell()` function and an initial stoploss of 10%? Not only is `trailing_stop` set to `false`, but 
 nowhere in my code is there any mention of a stoploss except for the `custom_sell()` function or the `stoploss = -0.10`.


![Screen Shot 2022-03-09 at 1 32 59 PM](https://user-images.githubusercontent.com/99715402/157413647-c2863989-5079-4f9d-9bb3-5fee10d70643.png)

**Comment by xmatthias** [comment]
can't say for sure just with the available details.

but you should be aware that stoploss settings can also come from the configuration - so having a look at the log (in particular the startup logs) can reveal different things

**Comment by ryhnm** [comment]
My startup logs are as follows:

`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'timeframe' with value in config file: 1h.`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USDT.`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: unlimited.`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'buy': 10, 'sell': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}.`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 1000}`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 1h`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.1`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False`
`2022-03-09 08:33:19,664 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: True`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: False`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'buy': 'market', 'sell': 'market', 'emergencysell': 'market', 'stoploss': 'market', 'stoploss_on_exchange': True}`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'buy': 'gtc', 'sell': 'gtc'}`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USDT`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: unlimited`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: []`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 150`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'buy': 10, 'sell': 10, 'exit_timeout_count': 0, 'unit': 'minutes'}`
`2022-03-09 08:33:19,665 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_sell_signal: True`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_only: False`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_buy_signal: False`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using sell_profit_offset: 0.0`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False`
`2022-03-09 08:33:19,666 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1`
`2022-03-09 08:33:19,666 - freqtrade.configuration.config_validation - INFO - Validating configuration ...`

This is the final issue I need to solve to get the algorithm live. Nowhere in the code, config or logs does it have any mention of a trailing stop loss, but it does suddenly pop up in backtesting.

**Comment by xmatthias** [comment]
you're using `use_custom_stoploss`  (which IS set to true).
If you don't add a custom implementation for `custom_stoploss` (which you should / have to if you use `use_custom_stoploss`) - then this will lead to trail with the original stoploss.

The current state (having `use_custom_stoploss=True` without implementation) does not really make much sense.
